src/commands/distribute/groups/update.ts

```diff
@@ -91,16 +100,32 @@ export default class UpdateDistributionGroupCommand extends AppCommand {
       addedTestersEmails = [];
     }
 
+    if (deletedTestersEmails.length !== testersToDeleteEmails.length || addedTestersEmails.length !== testersToAddEmails.length) {
+      out.text("Updating the list of testers was partially successful");
+    }
+
     let currentGroupName: string;
+    const options: { name?: string; isPublic?: boolean } = {};
     if (!_.isNil(this.newDistributionGroupName)) {
       debug("Renaming the distribution group");
-      currentGroupName = await this.renameDistributionGroup(client, app, this.newDistributionGroupName);
+      options.name = this.newDistributionGroupName;
+      currentGroupName = this.newDistributionGroupName;
     } else {
       currentGroupName = this.distributionGroup;
     }
 
-    if (deletedTestersEmails.length !== testersToDeleteEmails.length || addedTestersEmails.length !== testersToAddEmails.length) {
-      out.text("Updating the list of testers was partially successful");
+    if (this.makePublic) {
+      debug("Setting distribution group public status to true");
+      options.isPublic = true;
+    }
+
+    if (this.makePrivate) {
+      debug("Setting distribution group public status to false");
+      options.isPublic = false;
+    }
+
+    if (!_.isNil(options.name) || !_.isNil(options.isPublic)) {
+      await this.updateDistributionGroup(client, app, options);
```

Comment:
I am curious why option validation (like `this.makePublic && this.makePrivate`) is after the `updateDistributionGroup` calls :-P How does that work?

Reply:
Can you clarify your question? The call to `validateParameters` is right at the top of the `run` method.

Reply:
You might be confused by the lack of context provided by the GitHub changes view :(

Reply:
You are right. I was confused by the unfolded code parts... 😭